File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = float('-inf')
        curSum = 0
        for ele in nums:
            curSum += ele
            
            maxSum = max(maxSum, curSum)
            
            if curSum < 0: curSum = 0
            
        return maxSum
            
        

File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int maxi = INT_MIN;
        int curr = 0;
        
        for(int i : nums){
            curr += i;
            maxi = max(curr, maxi);
            if(curr <= 0) curr = 0;
        }
        
        return maxi;
        
    }
};